tutor/test_t1_59_ManageDistrictsSchoolsAndCourses.py

```diff
@@ -1241,49 +1474,128 @@ def test_admin_upload_a_student_roster_to_a_period_8358(self):
             )
         ).click()
 
-        self.admin.sleep(2)
+        # Create the course
+        self.admin.find(By.XPATH, "//a[text()='Add Course']").click()
+        self.admin.find(By.XPATH, "//input[@id='course_name']").send_keys(
+            course_name)
+        self.admin.find(By.XPATH, "//input[@value='Save']").click()
 
+        # Edit the course
+        self.admin.wait.until(
+            expect.visibility_of_element_located(
+                (By.XPATH, '//div[@class="pagination"]//a[not(@rel)][last()]')
+            )
+        ).click()
+        course = self.admin.driver.find_element(
+            By.XPATH,
+            "//div[@class='stats-card']" +
+            "//span[text()=' %s']" % course_name)
```

Comment:
switch the find statements to use the user function `find()` and `find_all()` instead of `driver.find_element()`, `driver.find_elements()` (the exception being if you perform a find from an element as opposed to the parent driver - ie `element.find_element()`)